### PR TITLE
Add comments explaining why use Multimap in DynamicFilters#extractSourceSymbols

### PR DESCRIPTION
**Pull request by tangjiangling**
Related https://github.com/trinodb/trino/pull/931/commits/d57a180437a7d07be593a0c4476f111af0965e43#r327738515

## Documentation

(x) No documentation is needed.

## Release notes

(x) No release notes entries required.

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/sql/planner/LocalDynamicFiltersCollector.java [hunk 1]
```diff
@@ -162,7 +162,7 @@ private TableSpecificDynamicFilter(Set<ColumnHandle> columnsCovered, List<Listen
             this.futuresLeft = predicateFutures.size();
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/sql/planner/LocalDynamicFiltersCollector.java:162]: commit message too long

File: core/trino-main/src/main/java/io/trino/sql/DynamicFilters.java [hunk 2]
```diff
@@ -133,6 +133,20 @@ public static ExtractResult extractDynamicFilters(Expression expression)
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/sql/DynamicFilters.java:133]: commit message too long

File: core/trino-main/src/main/java/io/trino/sql/DynamicFilters.java [hunk 3]
```diff
@@ -133,6 +133,20 @@ public static ExtractResult extractDynamicFilters(Expression expression)
 
     public static Multimap<DynamicFilterId, Descriptor> extractSourceSymbols(List<DynamicFilters.Descriptor> dynamicFilters)
     {
+        // Let's imagine if the probe-side is a union, and we need to match multiple
```
[the remark this answers is not in the document]
Comment (tangjiangling) [reply core/trino-main/src/main/java/io/trino/sql/DynamicFilters.java:136]: AFAIK we don't, and plan to cover the test in `TestDynamicFilter`.

[the remark this answers is not in the document]
Comment (sopel39) [reply core/trino-main/src/main/java/io/trino/sql/DynamicFilters.java:136]: Could you add such test that will fail without `Multimap` here?

[the remark this answers is not in the document]
Comment (tangjiangling) [reply core/trino-main/src/main/java/io/trino/sql/DynamicFilters.java:136]: Sure, will do.

**Comment by tangjiangling** [comment]
Convert to draft and I will re-activate it when I have time.

**Comment by mosabua** [comment]
:wave: @tangjiangling  - this PR has become inactive. We hope you are still interested in working on it. Please let us know, and we can try to get reviewers to help with that.

We're working on closing out old and inactive PRs, so if you're too busy or this has too many merge conflicts to be worth picking back up, we'll be making another pass to close it out in a few weeks.

**Comment by ebyhr** [comment]
Let me close as a stale PR. Please feel free to reopen if you want to continue.

**Pull request closed without merging**